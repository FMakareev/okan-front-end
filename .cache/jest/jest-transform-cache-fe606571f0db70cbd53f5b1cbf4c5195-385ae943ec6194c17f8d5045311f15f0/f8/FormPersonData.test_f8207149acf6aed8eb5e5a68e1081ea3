9862207550d14dfd6911d1dfc3914844
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));

var _reactRouterDom = require("react-router-dom");

var _StyledThemeProvider = require("../../../../styles/StyledThemeProvider");

var _FormPersonData = require("./FormPersonData");

it('FormPersonData: Рендерится без ошибок', function () {
  _reactTestRenderer["default"].create(_react["default"].createElement(_StyledThemeProvider.StyledThemeProvider, null, _react["default"].createElement(_FormPersonData.FormPersonData, null)));
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkZvcm1QZXJzb25EYXRhLnRlc3QuanMiXSwibmFtZXMiOlsiaXQiLCJyZW5kZXJlciIsImNyZWF0ZSJdLCJtYXBwaW5ncyI6Ijs7OztBQUFBOztBQUNBOztBQUNBOztBQUVBOztBQUNBOztBQUVBQSxFQUFFLENBQUMsdUNBQUQsRUFBMEMsWUFBTTtBQUNoREMsZ0NBQVNDLE1BQVQsQ0FDRSxnQ0FBQyx3Q0FBRCxRQUNFLGdDQUFDLDhCQUFELE9BREYsQ0FERjtBQUtELENBTkMsQ0FBRiIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBSZWFjdCBmcm9tICdyZWFjdCc7XHJcbmltcG9ydCByZW5kZXJlciBmcm9tICdyZWFjdC10ZXN0LXJlbmRlcmVyJztcclxuaW1wb3J0IHsgQnJvd3NlclJvdXRlciB9IGZyb20gJ3JlYWN0LXJvdXRlci1kb20nO1xyXG5cclxuaW1wb3J0IHsgU3R5bGVkVGhlbWVQcm92aWRlciB9IGZyb20gJy4uLy4uLy4uLy4uL3N0eWxlcy9TdHlsZWRUaGVtZVByb3ZpZGVyJztcclxuaW1wb3J0IHsgRm9ybVBlcnNvbkRhdGEgfSBmcm9tICcuL0Zvcm1QZXJzb25EYXRhJztcclxuXHJcbml0KCdGb3JtUGVyc29uRGF0YTog0KDQtdC90LTQtdGA0LjRgtGB0Y8g0LHQtdC3INC+0YjQuNCx0L7QuicsICgpID0+IHtcclxuICByZW5kZXJlci5jcmVhdGUoXHJcbiAgICA8U3R5bGVkVGhlbWVQcm92aWRlcj5cclxuICAgICAgPEZvcm1QZXJzb25EYXRhIC8+XHJcbiAgICA8L1N0eWxlZFRoZW1lUHJvdmlkZXI+LFxyXG4gICk7XHJcbn0pO1xyXG4iXX0=