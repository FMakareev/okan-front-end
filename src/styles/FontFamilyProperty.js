/**
 * @params {string} fontFamily - string name of the font name property in the theme
 * @params {object} theme
 * */
export const FontFamilyProperty = ({ fontFamily, theme }) => {
  try {
    if (fontFamily && theme.fontFamily) {
      return `font-family: '${theme.fontFamily[fontFamily]}';`;
    }
  } catch (e) {
    console.error(e);
  }
  return undefined;
};
export default FontFamilyProperty;
