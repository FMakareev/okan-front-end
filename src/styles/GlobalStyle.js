import { createGlobalStyle } from 'styled-components'


export const GlobalStyle = createGlobalStyle`
  *{
    box-sizing: border-box;
  }
  body, html, #app {
    padding: 0;
    margin: 0;
    width: 100%;
    min-height: 100vh;
    background-color: #fff;
  }
`;
