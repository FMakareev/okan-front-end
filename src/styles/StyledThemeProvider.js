import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'rebass';

/** Variants */
import { ButtonVariant } from './variants/buttonVariant';
import { ButtonSize } from './variants/buttonSize';
import { ButtonComment } from './variants/buttonComment';
import { TextVariant } from './variants/textVariant';
import { InputSize } from '@lib/styles/variants/InputSize';
import { InputVariant } from '@lib/styles/variants/InputVariant';

export const ColorPallet = {
  color0: '#FFFFFF',
  color1: '#E5E5E5',
  color2: '#DEDEDE',
  color3: '#4F4F4F',
  color4: '#848484',
  color5: '#007FAF',
  color6: '#006699',
  color7: '#00649C',
  color8: '#007CC4',
  color9: '#F97E63',
  color10: '#AEAEAE',
  color11: '#333333',
  color12: '#DF4624',
  color13: '#f2f2f2',
  color14: '#fff3c1',
};

const Space = [
  0,
  2,
  4,
  8,
  12,
  16,
  18,
  20,
  24,
  28,
  32,
  36,
  40,
  44,
  48,
  52,
  56,
  60,
  64,
  68,
  72,
  76,
  80,
];

const Weight = [300, 500, 700];

const Breakpoints = ['1024px', '1200px', '1440px'];

const boxShadow = [
  '2px 4px 8px 0px rgba(138,138,138,0.5)',
  '2px 4px 8px 0px rgba(138,138,138,0.5)',
  '2px 2px 4px 0px rgba(0,127,175,1)',
];

const textAlign = ['left', 'center', 'right'];

export const StyledThemeProvider = ({ children }) => (
  <Provider
    theme={{
      space: Space,
      fontSizes: Space,
      fontWeight: Weight,
      lineHeight: Space,
      boxShadow,
      borderRadius: Space,
      borderColor: ColorPallet,
      breakpoints: Breakpoints,
      colors: ColorPallet,
      variant: {
        buttons: ButtonVariant,
        buttonSize: ButtonSize,
        buttonComment: ButtonComment,
        text: TextVariant,
        inputSize: InputSize,
        inputVariant: InputVariant,
      },
      fontFamily: {
        primary300: 'Museo Sans 300',
        primary500: 'Museo Sans 500',
        primary700: 'Museo Sans 700',
        secondary: 'Circe Regular',
        secondaryBold: 'Circe Bold',
      },
      textAlign: textAlign,
    }}>
    {children}
  </Provider>
);

StyledThemeProvider.propTypes = {
  children: PropTypes.element.isRequired,
};

StyledThemeProvider.defaultProps = {
  children: 'children is empty',
};

export default StyledThemeProvider;
