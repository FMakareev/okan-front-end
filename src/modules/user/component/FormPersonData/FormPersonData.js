import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { reduxForm, Form, getFormValues } from 'redux-form';

/**Components*/
import Box from '../../../../components/Box/Box';
import Text from '../../../../components/Text/Text';

/** Styles property */
import { FontFamilyProperty } from '../../../../styles/styleProperty/FontFamilyProperty';
import { BorderColorProperty } from '../../../../styles/styleProperty/BorderColorProperty';
import { BorderRadiusProperty } from '../../../../styles/styleProperty/BorderRadiusProperty';

const BoxStyled = styled(Box)`
  border: 1px solid;
  ${props => BorderColorProperty({ ...props, borderColor: 'color4' })};
  ${props => BorderRadiusProperty({ ...props, borderRadius: '5px' })};
  padding: 10px;
`;

const TextStyled = styled(Text)`
  border-top: 2px solid;
  ${props => BorderColorProperty({ ...props, borderColor: 'color7' })};
  ${props => FontFamilyProperty({ ...props, fontFamily: 'secondary' })};
  padding-top: 4px;
  padding-bottom: 4px;
`;

export class FormPersonData extends Component {
  static propTypes = {
    /** last name user*/
    lastname: PropTypes.string /** first name user*/,
    firstname: PropTypes.string /** patronymic user*/,
    patronymic: PropTypes.string /** birth date user*/,
    birthdate: PropTypes.string /** position user*/,
    position: PropTypes.string /** phone user*/,
    phone: PropTypes.string /** email user*/,
    email: PropTypes.string,
    mb: PropTypes.string,
  };

  state = {};

  submit = value => {
    console.log('value', value);
  };
  render() {
    const {
      handleSubmit,
      initialValues: { lastname, firstname, patronymic, birthdate, position, phone, email },
    } = this.props;

    return (
      <Form onSubmit={handleSubmit(this.submit)}>
        <Text
          fontSize={6}
          lineHeight={8}
          color={'color7'}
          textAlign={'center'}
          mb={13}
          fontFamily={'primary500'}>
          Личные данные
        </Text>

        <BoxStyled>
          {lastname && (
            <Text fontSize={5} lineHeight={7} pl={4} pt={0} pb={2} fontFamily={'secondary'}>
              {lastname}
            </Text>
          )}

          {firstname && (
            <TextStyled fontSize={5} lineHeight={7} pl={4}>
              {firstname}
            </TextStyled>
          )}

          {patronymic && (
            <TextStyled fontSize={5} lineHeight={7} pl={4}>
              {patronymic}
            </TextStyled>
          )}

          {birthdate && (
            <TextStyled fontSize={5} lineHeight={7} pl={4}>
              {birthdate}
            </TextStyled>
          )}

          {position && (
            <TextStyled fontSize={5} lineHeight={7} pl={4}>
              {position}
            </TextStyled>
          )}

          {phone && (
            <TextStyled fontSize={5} lineHeight={7} pl={4}>
              {phone}
            </TextStyled>
          )}

          {email && (
            <TextStyled fontSize={5} lineHeight={7} pl={4} mb={-2}>
              {email}
            </TextStyled>
          )}
        </BoxStyled>
      </Form>
    );
  }
}

FormPersonData = connect(
  state => {
    ({ values: getFormValues('FormPersonData')(state) });
  },
  null,
)(FormPersonData);

FormPersonData = reduxForm({
  form: 'FormPersonData',
})(FormPersonData);

export default FormPersonData;
