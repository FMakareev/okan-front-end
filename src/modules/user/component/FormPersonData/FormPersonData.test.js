import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider as ProviderRedux } from 'react-redux';
import { createStore } from 'redux';

import { StyledThemeProvider } from '../../../../styles/StyledThemeProvider';
import { FormPersonData } from './FormPersonData';

const store = createStore(() => {});

it('FormPersonData: Рендерится без ошибок', () => {
  const renderWithProps = props => {
    const params = {
      initialValues: {
        lastname: 'one',
        firstname: 'two',
        patronymic: 'three',
        birthdate: '10.10.10',
        position: 'one',
        phone: '8-999-999-99-99',
        email: 'one',
      },
    };
    return renderer.create(
      <StyledThemeProvider>
        <ProviderRedux store={store}>
          <FormPersonData {...params} />
        </ProviderRedux>
      </StyledThemeProvider>,
    );
  };
  expect(renderWithProps({})).toMatchSnapshot();
});
