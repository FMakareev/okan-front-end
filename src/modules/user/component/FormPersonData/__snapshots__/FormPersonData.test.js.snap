// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormPersonData: Рендерится без ошибок 1`] = `
<div
  className="sc-bxivhb gPUjkv"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="sc-1jxwge6-0 sc-18ch56c-0 cRBGez"
      color="color11"
    >
      Личные данные
    </div>
    <div
      className="sc-1jxwge6-0 sc-8d86l7-0 hlhfAl"
    >
      <div
        className="sc-1jxwge6-0 sc-18ch56c-0 ggEpMk"
        color="color11"
        fontFamily="secondary"
        fontSize={5}
      >
        one
      </div>
      <div
        className="sc-1jxwge6-0 sc-18ch56c-0 sc-8d86l7-1 bfdiQg"
        color="color11"
        fontSize={5}
      >
        two
      </div>
      <div
        className="sc-1jxwge6-0 sc-18ch56c-0 sc-8d86l7-1 bfdiQg"
        color="color11"
        fontSize={5}
      >
        three
      </div>
      <div
        className="sc-1jxwge6-0 sc-18ch56c-0 sc-8d86l7-1 bfdiQg"
        color="color11"
        fontSize={5}
      >
        10.10.10
      </div>
      <div
        className="sc-1jxwge6-0 sc-18ch56c-0 sc-8d86l7-1 bfdiQg"
        color="color11"
        fontSize={5}
      >
        one
      </div>
      <div
        className="sc-1jxwge6-0 sc-18ch56c-0 sc-8d86l7-1 bfdiQg"
        color="color11"
        fontSize={5}
      >
        8-999-999-99-99
      </div>
      <div
        className="sc-1jxwge6-0 sc-18ch56c-0 sc-8d86l7-1 fBEXza"
        color="color11"
        fontSize={5}
      >
        one
      </div>
    </div>
  </form>
</div>
`;
